File: pkg/dto/plan.go
package dto

import (
	"github.com/KubeOperator/KubeOperator/pkg/constant"
	"github.com/KubeOperator/KubeOperator/pkg/model"
)

type Plan struct {
	model.Plan
	PlanVars   interface{} `json:"planVars"`
	RegionName string      `json:"regionName"`
	ZoneNames  []string    `json:"zoneNames"`
}

type PlanCreate struct {
	Name           string      `json:"name" validate:"koname,required"`
	Zones          []string    `json:"zones" validate:"required"`
	PlanVars       interface{} `json:"planVars" validate:"required"`
	DeployTemplate string      `json:"deployTemplate" validate:"required"`
	Projects       []string    `json:"projects" validate:"required"`
	Region         string      `json:"region" validate:"required"`
}

type PlanOp struct {
	Operation string `json:"operation" validate:"required"`
	Items     []Plan `json:"items" validate:"required"`
}

type PlanVmConfig struct {
	Name   string            `json:"name"`
	Config constant.VmConfig `json:"config"`
}

File: ui/src/app/business/deploy-plan/plan/plan.ts
import {BaseModel, BaseRequest} from '../../../shared/class/BaseModel';
import {Region} from '../region/region';
import {Zone} from '../zone/zone';

export class Plan extends BaseModel {
    name: string;
    zoneId: string;
    deployTemplate: string;
    vars: string;
    regionId: string;
    region: Region = new Region();
    zones: Zone[] = [];
    planVars: {} = {};
    regionName: string;
    zoneNames: string[] = [];
}

export class PlanCreateRequest extends BaseRequest {
    deployTemplate: string;
    vars: string;
    planVars: {} = {};
    // regionId: string;
    zone: string;
    zones: string [] = [];
    region: string;
    projects: string[] = [];
}

export class PlanVmConfig {
    name: string;
    config: VmConfig;
}

export class VmConfig {
    cpu: number;
    memory: number;
    disk: number;
}


